### PR TITLE
Document flowcontrol v1beta1 removal release

### PR DESCRIPTION
**Pull request by liggitt**
Updating documentation for flowcontrol/v1beta1 deprecation and replacement with v1beta2

### CODE REVIEW

**Comment by bradtopol** [comment]
Hi @liggitt,  Judging from your comments it looks like you want @MikeSpreitzer to be the technical reviewer for this.  @MikeSpreitzer if you feel the the PR is technically correct, just add a /lgtm  and I'll take care of the rest.  Thanks.

File: content/en/docs/reference/using-api/deprecation-guide.md [hunk 1]
```diff
@@ -24,12 +24,21 @@ deprecated API versions to newer and more stable API versions.
...
+* All existing persisted objects are accessible via the new API
+* No notable changes
+
 #### HorizontalPodAutoscaler {#horizontalpodautoscaler-v126}
 
 The **autoscaling/v2beta2** API version of HorizontalPodAutoscaler will no longer be served in v1.26.
 
 * Migrate manifests and API clients to use the **autoscaling/v2** API version, available since v1.23.
 * All existing persisted objects are accessible via the new API
+* No notable changes between **autoscaling/v2beta2** and **autoscaling/v2**
```
Comment (MikeSpreitzer) [inline content/en/docs/reference/using-api/deprecation-guide.md:41]: This looks like a logically separate change.

**Review by MikeSpreitzer — COMMENTED** [review]
The "flowcontrol" part looks good.
I'm not sure the other part should be in this PR, can't vet it myself.

File: content/en/docs/reference/using-api/deprecation-guide.md (hunk 1 above)
Comment (liggitt) [reply content/en/docs/reference/using-api/deprecation-guide.md:41]: oops, sorry, was left in my working tree as a follow-up to https://github.com/kubernetes/website/pull/30776#discussion_r765157906

Comment (liggitt) [reply content/en/docs/reference/using-api/deprecation-guide.md]: dropped this line from this PR

File: content/en/docs/reference/using-api/deprecation-guide.md [hunk 2]
```diff
@@ -24,6 +24,14 @@ deprecated API versions to newer and more stable API versions.
 
 The **v1.26** release will stop serving the following deprecated API versions:
 
+#### Flow control resources {#flowcontrol-v126}
```
Comment (sftim) [inline content/en/docs/reference/using-api/deprecation-guide.md:27]: nit:
```suggestion
#### Flow control resources {#flowcontrol-resources-v126}
```
to match fragment identifier `#webhook-resources-v122`

Comment (liggitt) [reply content/en/docs/reference/using-api/deprecation-guide.md:27]: done

**Review by MikeSpreitzer — APPROVED** [review]
/lgtm

**Pull request merged**